### PR TITLE
Reports page

### PR DESCRIPTION
**Pull request by AsalaKM**
**Reports Page**
Allow the manager to view all report by status, type or period

> By Selected Period
> By Selected Status
> By Selected Type

**Relates #17**

### CODE REVIEW

File: public/js/report.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+const solved = document.querySelectorAll('.solved');
+const notSolved = document.querySelectorAll('.not');
+const inProgress = document.querySelectorAll('.in');
+
+periodSection.style.display = 'none';
+switchType.style.display = 'none';
+switchStatus.style.display = 'none';
+reportSection.style.display = 'none';
+// loans.forEach((element) => {
+//   element.style.display = 'none';
```
Comment (FarahZaqout) [inline public/js/report.js]: No commented code.

File: src/model/database/dbBuild.sql [hunk 2]
```diff
@@ -22,7 +22,7 @@ CREATE TABLE problem_type (
 
 CREATE TABLE status_type (
     id SERIAL PRIMARY KEY,
-    status VARCHAR UNIQUE NOT NULL
+    status VARCHAR UNIQUE
```
Comment (FarahZaqout) [inline src/model/database/dbBuild.sql]: This change is not acceptable.

**Review by FarahZaqout — CHANGES_REQUESTED** [review]
2 Comments, plus, all sql statements should be in uppercase letters.

**Review by SalwaMugh — APPROVED** [review]

File: src/model/queries/ticket.js [hunk 3]
```diff
@@ -2,7 +2,14 @@ const dbConnection = require('../database/dbConnection');
 
 const getLatestTicket = () => {
   const sql = {
-    text: 'SELECT * FROM ticket ORDER BY ticket_no DESC',
+    text: 'SELECT * FROM ticket ORDER BY ticket_no DESC LIMIT 5;',
+  };
+  return dbConnection.query(sql);
+};
+
+const getAllTicket = () => {
```
Comment (FarahZaqout) [inline src/model/queries/ticket.js:10]: This function is the same as the one below it. This will crash the server for sure.

**Review by FarahZaqout — CHANGES_REQUESTED** [review]

File: src/controllers/reports.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+const {
+  getAllTicket, getTicketByPeriod, getLoans, getSupports, getNotSolved, getTicketByStatus,
+} = require('../model/queries/ticket');
+
+exports.get = async (req, res, next) => {
+  try {
+    const solved = 'solved';
+    const progress = 'in';
+    const ticket = await getAllTicket();
+    const tLoans = await getLoans();
```
Comment (FarahZaqout) [inline src/controllers/reports.js]: tLoans is not a good name.

**Review by FarahZaqout — CHANGES_REQUESTED** [review]

File: src/controllers/reports.js (hunk 4 above)
Comment (AsalaKM) [reply src/controllers/reports.js]: ok

**Review by OnsNader — APPROVED** [review]

**Pull request closed without merging**